app/models/rtu_date_range.rb

```diff
@@ -25,14 +27,14 @@ def search(query_body)
   end
 
   def extract_date_range(result)
-    facets = result["facets"]
-    return Date.current..Date.current if facets.nil?
-    stats = facets["stats"]
-    min, max = normalize(stats["min"]), normalize(stats["max"])
+    aggregations = result['aggregations']
+    return Date.current..Date.current if aggregations.nil?
+    stats = aggregations['stats']
+    min, max = normalize(stats['min']), normalize(stats['max'])
     min..max
   end
 
   def normalize(unixtime)
-    unixtime =~ /Infinity/ ? Date.current : DateTime.strptime(unixtime.to_s, '%Q').to_date
+    unixtime.blank? ? Date.current : DateTime.strptime(unixtime.to_s, '%Q').to_date
```

Comment:
I see that the old code had DateTime so its up to you but rubocop prefers Time over Date time.